src/librustc_metadata/encoder.rs

```diff
@@ -1684,6 +1707,11 @@ impl<'a, 'b, 'tcx> IndexBuilder<'a, 'b, 'tcx> {
                     let encode_info = IsolatedEncoder::encode_info_for_ty_param;
                     self.record(def_id, encode_info, (def_id, has_default));
                 }
+                hir::GenericParamKind::Const { .. } => {
+                    let def_id = self.tcx.hir().local_def_id(param.id);
+                    let encode_info = IsolatedEncoder::encode_info_for_const_param;
```

Comment:
Maybe open an issue about unifying the param encoding in metadata, the same way `GenericParamDef` has common fields?

Reply:
Opened https://github.com/rust-lang/rust/pull/58581.